modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js

```diff
@@ -251,7 +253,7 @@ class ContributorBuilder extends React.Component {
 
 											<CriteriaBuilder
 												criteria={criteria.criteriaMap}
-												editing={editingId === criteria.propertyKey}
+												editing={editing && editingId === criteria.propertyKey}
```

Comment:
At the beginning of the map, a `const editingCriteria = editing && editingId === criteria.propertyKey`.